huaweicloud/services/acceptance/ims/resource_huaweicloud_ims_obs_system_image_test.go

```diff
@@ -0,0 +1,395 @@
+package ims
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/chnsz/golangsdk/openstack/ims/v2/cloudimages"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+// This test case tests the creation of an ECS system image through an external image file in the OBS bucket.
+func TestAccObsSystemImage_basic(t *testing.T) {
+	var (
+		image        cloudimages.Image
+		rName        = acceptance.RandomAccResourceName()
+		rNameUpdate  = rName + "-update"
+		resourceName = "huaweicloud_ims_obs_system_image.test"
+		defaultEpsId = "0"
+		migrateEpsId = acceptance.HW_ENTERPRISE_PROJECT_ID_TEST
+	)
+
+	rc := acceptance.InitResourceCheck(
+		resourceName,
+		&image,
+		getImsImageResourceFunc,
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			// This test case need setting a non default enterprise project ID.
+			acceptance.TestAccPreCheckEpsID(t)
+			// This test requires ensuring that there is an image file in the OBS bucket.
+			acceptance.TestAccPreCheckImsImageUrl(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccObsSystemImage_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "image_url", acceptance.HW_IMS_IMAGE_URL),
+					resource.TestCheckResourceAttr(resourceName, "min_disk", "60"),
+					resource.TestCheckResourceAttr(resourceName, "description", "terraform description test"),
+					resource.TestCheckResourceAttr(resourceName, "os_type", "Windows"),
+					resource.TestCheckResourceAttr(resourceName, "os_version", "Windows Server 2019 Standard 64bit"),
```

Comment:
It is recommended not to specify the specific type or version of the OS, because the image file in OBS is uncertain.

Reply:
fixed.

Reply:
This test requires specifying these parameters